### ISSUE TITLE
List block: when removing an empty list item, should not merge two list items

### ISSUE THREAD

**Issue by v18** [WordPress/gutenberg#12398]
**Describe the bug**
When I press the 'delete' key on an empty list item, the item below the empty one gets merged with the one above.

A gif demo:

![remove-empty-list-item](https://user-images.githubusercontent.com/11487924/49179928-6217d700-f321-11e8-86fc-ce00605f202d.gif)

**To Reproduce**
1. Make a new list gutenberg block
2. Add some items
3. Make one of the items empty (no text, but still with a number / bullet)
4. Put cursor at the empty list item
5. Press 'delete' on your keyboard
6. See the two items around the empty list item merged

**Expected behavior**

Starting from this list:

<img width="149" alt="list-before-delete" src="https://user-images.githubusercontent.com/11487924/49180035-aacf9000-f321-11e8-85eb-7def8486b181.png">

```
1. Apples
2. Oranges
3. 
4. Limes
5. Lemons
```

I expected to see this after pressing 'delete' on item 3:

<img width="140" alt="list-after-delete" src="https://user-images.githubusercontent.com/11487924/49180065-b9b64280-f321-11e8-8ff3-d62fbcd239aa.png">

```
1. Apples
2. Oranges
4. Limes
5. Lemons
```

**Comment by designsimply** [comment]
Tested and confirmed that deleting an empty list item merges the list items before and after it using WordPress 4.9.8 and Gutenberg 4.5.1 `master` @ `e4fd51cb5` using Firefox 63.0.3 on macOS 10.13.6.

**Comment by iseulde** [comment]
Thanks for the report!

**Comment by ZebulanStanphill** [comment]
I can confirm this is still happening in Gutenberg 4.6.1.

**Comment by youknowriad** [comment]
I can't reproduce, I suppose it was fixed by #13697

**Comment by iseulde** [comment]
Thanks for testing. Yes, I think it was TinyMCE and RichText both acting at the same time.
